### DISCUSSION TITLE
Modern Monitoring

### DISCUSSION THREAD

**Discussion by brentsec** [BretFisher/ama#162] [Q&A]
Hey Bret!

First, I just want to say thanks for your Docker Mastery course. I learned a lot from supplementing your course along with your YouTube livestreams and content

So one thing I feel that is missing from a lot of courses, and honestly in general from DevOps/developers around the world, are good resources for extremely critical production ready services like logging and monitoring. Obviously Docker makes this way easier since we can just spinup an ELK+Prometheus stack for example.

This isn't necessarily super sexy DevOps stuff, but in my professional experience it's something that is always mandatory, and tends to be the difference between someone who can be proactive in fixing problems rather than reactive. This is crucial in modern DevOps. I would also be extremely interested in seeing some best practices for using something like an ELK+Prometheus stack to not only monitor apps and services in something like a Docker Swarm/Kubernetes cluster, but also monitoring the swarm/cluster itself. 

If you have any resources you could point me too about this stuff, that would be great.  However, I think a course or guide with best practices on how to accomplish this would be awesome!

I can't even imagine how hard and time consuming creating quality content like you have is, but I just wanted to say I really appreciate your hard work. Thanks again, and keep it up! :)

**Comment by BretFisher** [comment]
Thanks! I'm glad you're enjoying my content, and I agree we need more intermediate and advanced education out there.

The main issue is that anything beyond basics in a topic sells at a fraction of 101 style courses (a little insider secret), and combined with the fact that those courses are harder to make, it means there will be limited options until someone can make it economical to produce that content.

I have lots of ideas like that on my radar, as "ops stuff" is my gig.

There are Prometheus books, and lots of conference talks that will get you certain answers and techniques, but I don't know of a one-stop-shop to teach you the tooling, features, as well as all the day-to-day app-specific stuff like "top 5 monitoring techniques with Prometheus for postgres" for example... which is useful I'm sure but very specific to a small audience.

Thanks for the thoughts, and remember to share what you learn, so you can help fill this gap!
